omegaconf/basecontainer.py

```diff
@@ -88,10 +88,39 @@ def __repr__(self) -> str:
 
     # Support pickle
     def __getstate__(self) -> Dict[str, Any]:
-        return self.__dict__
+        from omegaconf import DictConfig
+
+        dict_ = copy.copy(self.__dict__)
+        if sys.version_info < (3, 7, 0):  # pragma: no cover
```

Comment:
Let's do this unconditionally (for all Python versions).
By conditioning it, you end up with two different file formats, depending on the version that saved.
This makes it tricky if you save in 3.6 and load in 3.7.